libsession/src/main/java/org/session/libsession/snode/OnionRequestAPI.kt

```diff
@@ -190,8 +191,19 @@ object OnionRequestAPI {
                 if (unusedSnodes.count() < pathSnodeCount) { throw InsufficientSnodesException() }
                 // Don't test path snodes as this would reveal the user's IP to them
                 guardSnodes.minus(reusableGuardSnodes).map { guardSnode ->
-                    val result = listOf( guardSnode ) + (0 until (pathSize - 1)).map {
-                        val pathSnode = unusedSnodes.getRandomElement()
+                    val result = listOf( guardSnode ) + (0 until (pathSize - 1)).mapIndexed() { index, _ ->
+                        var pathSnode = unusedSnodes.getRandomElement()
+
+                        // we want to make sure the last node in the path is above version 2.8.0
+                        // to help with an issue that will disappear once the nodes are all updated
+                        if(index == pathSize - 2) {
+                            // because we are now grabbing the whole node pool there should always
+                            // be a node that is above version 2.8.0
+                            while(Util.compareVersions(pathSnode.version, "2.8.0") < 0) {
```

Comment:
One final thing, sometimes I'll put `emergencyExit` conditions on while loops like this just in case for whatever god-forsaken reason we end up trapped. Like if we go through the loop 1000 times then nah (maybe we have a small pool size, maybe the RNG is giving us the same node over and over, maybe anything..). Up to you, though.